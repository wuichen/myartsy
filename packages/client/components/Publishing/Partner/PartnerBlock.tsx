import { unica } from "Assets/Fonts"
import { pMedia } from "Components/Helpers"
import React from "react"
import track, { TrackingProp } from "react-tracking"
import styled from "styled-components"
import { resize } from "Utils/resizer"

interface Props {
  logo: string
  url: string
  tracking?: TrackingProp
  trackingData?: any
}

@track()
export class PartnerBlock extends React.Component<Props, null> {
  static defaultProps = {}

  @track(props => ({
    action: "Click",
    type: "external_link",
    destination_path: props.url,
  }))
  onPartnerClick(event) {
    // noop
  }

  render() {
    const { logo, url } = this.props
    const resized_logo = resize(logo, { width: 240 })
    const image = <img src={resized_logo} />

    return (
      <PartnerBlockContainer className="PartnerBlock">
        {logo && <Title>Presented in Partnership with</Title>}
        {image && (
          <ImageContainer>
            {url ? (
              <a
                href={url}
                target="_blank"
                onClick={this.onPartnerClick.bind(this)}
              >
                {image}
              </a>
            ) : (
              image
            )}
          </ImageContainer>
        )}
      </PartnerBlockContainer>
    )
  }
}

export const PartnerBlockContainer = styled.div`
  display: block;
`

export const ImageContainer = styled.div`
  img {
    max-width: 240px;
    max-height: 40px;
    object-fit: contain;
    object-position: left;
  }
`

const Title = styled.div`
  ${unica("s16", "medium")};
  margin-bottom: 20px;

  ${pMedia.sm`
    ${unica("s14", "medium")}
  `};
`
