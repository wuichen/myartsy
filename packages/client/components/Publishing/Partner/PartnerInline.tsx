import { pMedia } from "Components/Helpers"
import Icon from "Components/Icon"
import { IconPlus } from "Components/Publishing/Icon/IconPlus"
import React from "react"
import track from "react-tracking"
import styled from "styled-components"
import { resize } from "Utils/resizer"

interface Props {
  color?: string
  logo: string
  url: string
  trackingData?: any
  margin?: string
}

interface DivProps extends React.HTMLProps<HTMLDivElement> {
  margin: string
}

@track()
export class PartnerInline extends React.Component<Props, null> {
  static defaultProps = {
    margin: "0px",
  }

  @track(props => ({
    action: "Click",
    type: "external_link",
    destination_path: props.url,
  }))
  onPartnerClick(event) {
    // noop
  }

  render() {
    const { color, logo, url, margin } = this.props
    const resized_logo = logo ? resize(logo, { width: 240 }) : ""

    return (
      <PartnerInlineContainer margin={margin} className="PartnerInline">
        <a href="/">
          <Icon name="logo" color={color ? color : "black"} fontSize="32px" />
        </a>
        {resized_logo && (
          <>
            <IconPlus color={color} />
            <a
              href={url}
              target="_blank"
              onClick={this.onPartnerClick.bind(this)}
            >
              <img src={resized_logo} />
            </a>
          </>
        )}
      </PartnerInlineContainer>
    )
  }
}

const PartnerInlineContainer = styled.div`
  z-index: 1;
  display: flex;
  align-items: center;
  margin: ${(props: DivProps) => props.margin};

  img {
    max-height: 34px;
    vertical-align: middle;
    /* stylelint-disable-next-line */
    -ms-interpolation-mode: bicubic;
  }

  ${IconPlus} {
    margin: 0 20px;
  }

  ${Icon} {
    margin: 0;
    vertical-align: middle;
  }
  ${pMedia.sm`
    img {
      max-height: 24px;
    }
    ${Icon} {
      font-size: 24px;
    }
    ${IconPlus} {
      margin: 0 10px;
    }
  `};
`
