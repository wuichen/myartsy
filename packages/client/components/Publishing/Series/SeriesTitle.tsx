import { media, Sans } from "@artsy/palette"
import { ArticleData } from "Components/Publishing/Typings"
import React from "react"
import styled from "styled-components"
import { ImageContainer, PartnerBlock } from "../Partner/PartnerBlock"

interface SeriesTitleProps extends React.HTMLProps<HTMLDivElement> {
  article?: ArticleData
  color?: string
  editTitle?: any
}

export const SeriesTitle: React.SFC<SeriesTitleProps> = ({
  article,
  color = "black",
  editTitle,
}) => {
  const { sponsor, title } = article

  return (
    <SeriesTitleContainer color={color}>
      <Sans size={["10", "12", "16"]} element="h1" mb={["2", "3"]}>
        {editTitle ? editTitle : title}
      </Sans>

      {sponsor && (
        <PartnerBlock
          logo={sponsor.partner_light_logo}
          url={sponsor.partner_logo_link}
          trackingData={{
            type: "external link",
            destination_path: sponsor.partner_logo_link,
          }}
        />
      )}
    </SeriesTitleContainer>
  )
}

export const SeriesTitleContainer = styled.div<{ color: string }>`
  color: ${props => props.color};
  text-align: center;

  ${ImageContainer} {
    padding-top: 5px;
    padding-bottom: 40px;

    img {
      margin-left: auto;
      margin-right: auto;
    }
  }

  ${media.md`
    ${ImageContainer} {
      padding-bottom: 0;
    }
  `};
`
