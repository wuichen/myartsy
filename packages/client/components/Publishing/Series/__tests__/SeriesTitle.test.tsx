import {
  SeriesArticle,
  SeriesArticleSponsored,
} from "Components/Publishing/Fixtures/Articles"
import { EditableChild } from "Components/Publishing/Fixtures/Helpers"
import { PartnerBlock } from "Components/Publishing/Partner/PartnerBlock"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { SeriesTitle } from "../SeriesTitle"

it("renders a series title properly", () => {
  const component = renderer
    .create(<SeriesTitle article={SeriesArticle} />)
    .toJSON()
  expect(component).toMatchSnapshot()
})

it("renders a sponsored series title properly", () => {
  const component = renderer
    .create(<SeriesTitle article={SeriesArticleSponsored} />)
    .toJSON()
  expect(component).toMatchSnapshot()
})

it("renders a series with children properly", () => {
  const component = renderer
    .create(
      <SeriesTitle article={SeriesArticle} editTitle={EditableChild("title")} />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

it("Renders partner block for a sponsored series", () => {
  const component = mount(<SeriesTitle article={SeriesArticleSponsored} />)
  expect(component.find(PartnerBlock).length).toBe(1)
})

it("Does not render partner block for an unsponsored series", () => {
  const component = mount(<SeriesTitle article={SeriesArticle} />)
  expect(component.find(PartnerBlock).length).toBe(0)
})

it("Renders children if present", () => {
  const component = mount(
    <SeriesTitle article={SeriesArticle} editTitle={EditableChild("title")} />
  )
  expect(component.text()).toMatch("Child title")
})
