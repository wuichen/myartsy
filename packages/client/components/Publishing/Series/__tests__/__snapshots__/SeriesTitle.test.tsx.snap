// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a series title properly 1`] = `
.c1 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 42px;
  line-height: 50px;
  margin-bottom: 8px;
}

.c0 {
  color: black;
  text-align: center;
}

.c0 .c2 {
  padding-top: 5px;
  padding-bottom: 40px;
}

.c0 .c2 img {
  margin-left: auto;
  margin-right: auto;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 60px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    font-size: 100px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    line-height: 66px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    line-height: 104px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    margin-bottom: 16px;
  }
}

@media (max-width:900px) {
  .c0 .c2 {
    padding-bottom: 0;
  }
}

<div
  className="c0"
  color="black"
>
  <h1
    className="c1"
    fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
    fontSize={
      Array [
        "42px",
        "60px",
        "100px",
      ]
    }
  >
    The Future of Art
  </h1>
</div>
`;

exports[`renders a series with children properly 1`] = `
.c1 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 42px;
  line-height: 50px;
  margin-bottom: 8px;
}

.c0 {
  color: black;
  text-align: center;
}

.c0 .c2 {
  padding-top: 5px;
  padding-bottom: 40px;
}

.c0 .c2 img {
  margin-left: auto;
  margin-right: auto;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 60px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    font-size: 100px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    line-height: 66px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    line-height: 104px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    margin-bottom: 16px;
  }
}

@media (max-width:900px) {
  .c0 .c2 {
    padding-bottom: 0;
  }
}

<div
  className="c0"
  color="black"
>
  <h1
    className="c1"
    fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
    fontSize={
      Array [
        "42px",
        "60px",
        "100px",
      ]
    }
  >
    <div>
      Child 
      title
    </div>
  </h1>
</div>
`;

exports[`renders a sponsored series title properly 1`] = `
.c1 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 42px;
  line-height: 50px;
  margin-bottom: 8px;
}

.c2 {
  display: block;
}

.c5 img {
  max-width: 240px;
  max-height: 40px;
  object-fit: contain;
  object-position: left;
}

.c3 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  margin-bottom: 20px;
}

.c0 {
  color: black;
  text-align: center;
}

.c0 .c4 {
  padding-top: 5px;
  padding-bottom: 40px;
}

.c0 .c4 img {
  margin-left: auto;
  margin-right: auto;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 60px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    font-size: 100px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    line-height: 66px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    line-height: 104px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    margin-bottom: 16px;
  }
}

@media (max-width:720px) {
  .c3 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
  }
}

@media (max-width:900px) {
  .c0 .c4 {
    padding-bottom: 0;
  }
}

<div
  className="c0"
  color="black"
>
  <h1
    className="c1"
    fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
    fontSize={
      Array [
        "42px",
        "60px",
        "100px",
      ]
    }
  >
    The Future of Art
  </h1>
  <div
    className="PartnerBlock c2"
  >
    <div
      className="c3"
    >
      Presented in Partnership with
    </div>
    <div
      className="c4 c5"
    >
      <a
        href="http://artsy.net"
        onClick={[Function]}
        target="_blank"
      >
        <img
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FGEu3iYW6CQhbVxsjpOYwZQ%252FGKL_Wort-Bildmarke_negative_rgb%2B2.png&width=240&quality=80"
        />
      </a>
    </div>
  </div>
</div>
`;
